client/src/app/core/actions/poll-action.ts

```diff
@@ -0,0 +1,158 @@
+    export const UPDATE = 'poll.update';
+    export const DELETE = 'poll.delete';
+
+    export const PUBLISH = 'poll.publish';
+    export const RESET = 'poll.reset';
+    export const START = 'poll.start';
+    export const STOP = 'poll.stop';
+    export const ANONYMIZE = 'poll.anonymize';
+    export const VOTE = 'poll.vote';
+    export const UPDATE_OPTION = 'option.update';
+
+    interface UserIdentifiable {
+        user_id: Id;
+    }
+
+    type YNA = 'Y' | 'N' | 'A';
+
+    interface Option {
+        // Exactly one of text and content_object_id must be given
+        text?: string;
+        content_object_id?: Fqid;
+    }
+
+    export interface AnalogOption extends Option {
+        // Only for type==analog, optional votes can be given
+        Y?: FixedPointNumber; // Y, YN, YNA mode
+        N?: FixedPointNumber; // N, YN mode
+        A?: FixedPointNumber; // YNA mode
+    }
+
+    export interface NonAnalogOption extends Option {}
+
+    interface AnalogVotesPayload {
+        votesvalid?: FixedPointNumber;
+        votesinvalid?: FixedPointNumber;
+        votescast?: FixedPointNumber;
+    }
+
+    interface AnalogGlobalAmountPayload {
+        // Only for type==analog, optional votes can be given
+        amount_global_yes?: FixedPointNumber;
+        amount_global_no?: FixedPointNumber;
+        amount_global_abstain?: FixedPointNumber;
+    }
+
+    interface PartialCreatePayload extends HasMeetingId {
+        // Required
+        title: string;
+        type: string;
+        pollmethod: string;
+
+        options: Option[]; // must have at least one entry.
+
+        // Optional
+        content_object_id?: Fqid;
+        description?: string;
+        min_votes_amount?: number;
+        max_votes_amount?: number;
+        global_yes?: boolean;
+        global_no?: boolean;
+        global_abstain?: boolean;
+        onehundred_percent_base?: string;
+        majority_method?: string;
+    }
+
+    interface PartialUpdatePayload {
+        // Optional, every state
+        title?: string;
+        description?: string;
+        onehundred_percent_base?: string;
+        majority_method?: string;
+    }
+
+    interface PartialUpdateCreatedPollPayload {
+        // Optional, only if state == created
+        pollmethod?: string;
+        type?: string;
+        min_votes_amount?: number;
+        max_votes_amount?: number;
+        allow_multiple_votes_per_candidate?: boolean;
+        global_yes?: boolean;
+        global_no?: boolean;
+        global_abstain?: boolean;
+    }
+
+    interface PartialUpdateAnalogPayload extends AnalogVotesPayload, AnalogGlobalAmountPayload {
+        // type==analog, every state
+    }
+    export interface CreateAnalogPollPayload
+        extends PartialCreatePayload,
+            AnalogVotesPayload,
+            AnalogGlobalAmountPayload {
+        options: AnalogOption[];
+
+        // Only for type==analog
+        publish_immediately?: boolean;
```

Comment:
This is also valid for updating analog polls (e.g. if you give only `votesvalid` and `votescast` and do not do an `option.update`, you still can publish the poll)